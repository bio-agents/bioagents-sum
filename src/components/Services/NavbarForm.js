import React from 'react'
import { Field, formValueSelector, reduxForm } from 'redux-form'
import buildActionCreators from '../../helpers/buildActionCreators'
import * as ActionTypes from '../../constants/actionTypes'
import { connect } from 'react-redux'
import { Button, Form } from 'react-bootstrap'
import NavbarTextField from '../common/NavbarTextField'
import { getActiveCollection } from '../../selectors/collectionSelector'

class NavbarForm extends React.PureComponent {
  setCollection = event => {
    event.preventDefault()
    const { setCollection, servicesAndCitationsFetchCancel, enteredCollection, activeCollection } = this.props
    if (activeCollection !== enteredCollection) {
      servicesAndCitationsFetchCancel()
      setCollection({collection: enteredCollection, userEnteredCollection: true})
    }
  }

  render () {
    return (
      <Form onSubmit={this.setCollection}>
        <Field name='enteredCollection' component={NavbarTextField} placeholder='Enter collection' />
        {' '}
        <Button type='submit'>{'Submit'}</Button>
      </Form>
    )
  }
}

NavbarForm = reduxForm({
  form: 'collectionForm',
})(NavbarForm)

export default connect(state => {
  const selector = formValueSelector('collectionForm')

  return {
    enteredCollection: selector(state, 'enteredCollection'),
    initialValues: { enteredCollection: '' },
    activeCollection: getActiveCollection(state),
  }
}, buildActionCreators({
  servicesAndCitationsFetchCancel: ActionTypes.SERVICES_AND_CITATIONS_FETCH_CANCEL,
  setCollection: ActionTypes.SET_COLLECTION,
}))(NavbarForm)
