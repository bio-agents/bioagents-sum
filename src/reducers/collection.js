import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { configCollection } from '../bioagentsSum/common/helperFunctions'

const initialState = {
  activeCollection: '',
  userEnteredCollection: false,
}

export const collection = (state = initialState, action) => {
  const { type, payload } = action

  switch (type) {
    case Actions.SET_COLLECTION:
      return R.evolve({
        activeCollection: R.always(payload.collection !== '' ? payload.collection : configCollection),
        userEnteredCollection: R.always(payload.collection !== '' ? payload.userEnteredCollection : false),
      })(state)

    default:
      return state
  }
}
