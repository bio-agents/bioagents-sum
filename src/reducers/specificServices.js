import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { pickData } from '../bioagentsSum/services/index'

const initialState = {
  count: 0,
  list: [],
  persistExpiresAt: '',
  serviceLoading: false,
  serviceLoaded: false,
  citationsLoading: false,
  citationsLoaded: false,
}

export const specificServicesWithName = (servicesName = '') =>
  (state = initialState, action) => {
    const { type, payload } = action

    if (payload && payload.name !== servicesName) return state

    switch (type) {
      case Actions.SERVICES_FETCH:
        return R.evolve({
          serviceLoading: R.T,
          serviceLoaded: R.F,
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.SERVICES_FETCH_SUCCESS: {
        const { count, list } = payload.service
        let finalList = pickData(list)

        if (state.list.length !== 0) {
          const oldListByIds = R.groupBy(R.prop('id'), state.list)
          finalList = R.map(agent => R.mergeDeepLeft(agent, oldListByIds[agent.id] ? oldListByIds[agent.id][0] : {}), finalList)
        }

        return R.evolve({
          count: R.always(count),
          list: R.always(finalList),
          serviceLoading: R.F,
          serviceLoaded: R.T,
          citationsLoading: R.T,
          persistExpiresAt: R.always(Date.now() + 600000), // 86400000 ms == 24 hours
        })(state)
      }

      case Actions.SERVICES_FETCH_FAILURE: {
        return R.evolve({
          serviceLoading: R.F,
          serviceLoaded: R.F,
        })(state)
      }

      case Actions.CITATIONS_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
          citationsLoading: R.F,
          citationsLoaded: R.T,
        })(state)
      }

      case Actions.CITATIONS_FETCH_FAILURE:
        return R.evolve({
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.UPTIME_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
        })(state)
      }

      case Actions.UPTIME_FETCH_FAILURE: {
        // todo
      }

      default:
        return state
    }
  }
