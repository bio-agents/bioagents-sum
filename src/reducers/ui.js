import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'

const initialState = {
  showReportPage: false,
}

export const ui = (state = initialState, action) => {
  const { type } = action

  switch (type) {
    case Actions.REPORT_PAGE_CHANGE_VISIBILITY:
      return R.evolve({
        showReportPage: R.not,
      })(state)

    default:
      return state
  }
}
