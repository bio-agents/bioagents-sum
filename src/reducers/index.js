import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import { reducer as reduxFormReducer } from 'redux-form'
import { specificServicesWithName } from './specificServices'
import * as R from 'ramda'
import { ui } from './ui'
import { collection } from './collection'
import { getServicesNames } from '../bioagentsSum/common/helperFunctions'

const bioagentsSumApp = () => {
  const reducers = R.map(specificServicesWithName, getServicesNames)
  const servicesReducers = R.zipObj(getServicesNames, reducers)

  return combineReducers(R.merge(servicesReducers, {
    ui,
    collection,
    router: routerReducer,
    form: reduxFormReducer,
  }))
}

export default bioagentsSumApp
