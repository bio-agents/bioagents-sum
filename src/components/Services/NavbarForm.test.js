import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'
import reducers from '../../reducers'
import { Provider } from 'react-redux'
import NavbarForm from './NavbarForm'

const store = createStore(reducers(), {})

it('Renders MatrixCellWithLink with link and name correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <NavbarForm />
    </Provider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
