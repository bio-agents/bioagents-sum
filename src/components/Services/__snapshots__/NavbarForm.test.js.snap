exports[`test Renders MatrixCellWithLink with link and name correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      id={undefined}
      name="enteredCollection"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Enter collection"
      type="text"
      value="" />
  </div>
   
  <button
    className="btn btn-default"
    disabled={false}
    type="submit">
    Submit
  </button>
</form>
`;
